sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/LabelClassifyAsyncClient.java

```diff
@@ -0,0 +1,463 @@
+                                new ClassifyDocumentOperationDetail();
+                            ClassifyDocumentOperationDetailPropertiesHelper.setOperationId(operationDetail,
+                                parseOperationId(responseBase.getDeserializedHeaders().getOperationLocation()));
+                            return operationDetail;
+                        })),
+                pollingOperationTextJob(
+                    operationId -> service.jobStatusWithResponseAsync(operationId,
+                        finalIncludeStatistics, null, null, finalContext)),
+                cancelOperationTextJob(
+                    operationId -> service.cancelJobWithResponseAsync(operationId, finalContext)),
+                fetchingOperationIterable(
+                    operationId -> Mono.just(new ClassifyDocumentPagedIterable(
+                        getClassifyDocumentPagedFlux(operationId, null, null,
+                            finalIncludeStatistics, finalContext))))
+            );
+        } catch (RuntimeException ex) {
+            return PollerFlux.error(ex);
+        }
+    }
+
+    ClassifyDocumentPagedFlux getClassifyDocumentPagedFlux(
+        UUID operationId, Integer top, Integer skip, boolean showStats, Context context) {
+        return new ClassifyDocumentPagedFlux(
+            () -> (continuationToken, pageSize) ->
+                getPagedResult(continuationToken, operationId, top, skip, showStats, context).flux());
+    }
+
+    Mono<PagedResponse<ClassifyDocumentResultCollection>> getPagedResult(String continuationToken,
+        UUID operationId, Integer top, Integer skip, boolean showStats, Context context) {
+        try {
+            if (continuationToken != null) {
+                final Map<String, Object> continuationTokenMap = parseNextLink(continuationToken);
+                final Integer topValue = (Integer) continuationTokenMap.getOrDefault("$top", null);
+                final Integer skipValue = (Integer) continuationTokenMap.getOrDefault("$skip", null);
+                final Boolean showStatsValue = (Boolean) continuationTokenMap.getOrDefault(showStats, false);
+                return service.jobStatusWithResponseAsync(operationId, showStatsValue, topValue, skipValue,
+                    context)
+                    .map(this::toSingleClassifyDocumentResultCollectionPagedResponse)
+                    .onErrorMap(Utility::mapToHttpResponseExceptionIfExists);
+            } else {
+                return service.jobStatusWithResponseAsync(operationId, showStats, top, skip, context)
+                    .map(this::toSingleClassifyDocumentResultCollectionPagedResponse)
+                    .onErrorMap(Utility::mapToHttpResponseExceptionIfExists);
+            }
+        } catch (RuntimeException ex) {
+            return monoError(LOGGER, ex);
+        }
+    }
+
+    private PagedResponse<ClassifyDocumentResultCollection> toSingleClassifyDocumentResultCollectionPagedResponse(
```

Comment:
it says toSingleClassifyDocumentResultCollectionPagedResponse but there seems to be code on L310 that has to do with multi label results... sorry don't know enough java to know if this is normal.

Reply:
Good catch, it should be re-named to `toClassifyDocumentResultCollectionPagedResponse`